rabbitmq/datadog_checks/rabbitmq/rabbitmq.py

```diff
@@ -531,12 +531,15 @@ def _get_metrics(self, data, object_type, custom_tags):
             # Walk down through the data path, e.g. foo/bar => d['foo']['bar']
             root = data
             keys = attribute.split('/')
-            for path in keys[:-1]:
-                root = root.get(path, {})
+            value = None
+            try:  # In RabbitMQ 3.1.x queue_totals is a list instead of a dict
+                for path in keys[:-1]:
+                    root = root.get(path, {})
+                value = root.get(keys[-1], None)
+            except AttributeError as e:
```

Comment:
Can we instead check the type to avoid any error?

Reply:
We could do something like this, but I consider it a bit uglier and this only happens in rabbit 3.1 (2013). Once we do inventories for rabbit we can just ignore queue_totals for that version

```
for path in keys[:-1]:
    if not isinstance(root, dict):
        break
    root = root.get(path, {})
root.get(keys[-1], None) if isinstance(root, dict) else None
```